src/frontend/web_application/src/scenes/RemoteIdentitySettings/components/RemoteIdentityOauth/index.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Trans, withI18n } from '@lingui/react';
+import { Confirm, Button } from '../../../../components';
+import { REMOTE_IDENTITY_STATUS_ACTIVE, REMOTE_IDENTITY_STATUS_INACTIVE, Identity } from '../../../../modules/remoteIdentity';
+import RemoteIdentityDetails from '../RemoteIdentityDetails';
+
+@withI18n()
+class RemoteIdentitOauth extends Component {
+  static propTypes = {
+    i18n: PropTypes.shape({}).isRequired,
+    className: PropTypes.string,
+    remoteIdentity: PropTypes.shape({}).isRequired,
+    onChange: PropTypes.func.isRequired,
+    onDelete: PropTypes.func.isRequired,
+  };
+  static defaultProps = {
+    className: undefined,
+  };
+  state = {
+    remoteIdentity: {
+      identifier: '',
+      active: true,
+    },
+  };
+
+  componentWillMount() {
```

Comment:
I always used `componentWillMount` as an alternative to initiating the state in the constructor.

Don't forget about this article, https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

_(The name `getDerivatedStateFromProps` looks safe but introduces an anti-pattern and eventual bugs.)_

I think the less worst solution is declare the state directly:

```
class Foo extends Component {
  // optional initial state
  static initialState = { foo: 'bar' }
  state = this.constructor.getStateFromProps(this.props, this.constructor.initialState)

  static getStateFromProps = (props, prevState = {}) => ({
    ...prevState,
    // extract state from props
    ...
  }
  
  ...
}
```

other way, (but I don't like it since I usually forgot the `super(props)`:

```
class Foo extends Component {
  constructor(props) {
    super(props);
    // optional initial state
    const initialState = { foo: 'bar' }
    this.state = this.constructor.getStateFromProps(this.props, this.constructor.initialState)
  }

  static getStateFromProps = (props, prevState = {}) => ({
    ...prevState,
    // extract state from props
    ...
  }
  
  ...
}
```